### PR TITLE
Support partial builds

### PR DESCRIPTION
**Pull request by rth**
Following discussion in https://github.com/iodide-project/pyodide/issues/553 this adds the support of partial builds. From the added documentation,

> To build a subset of available packages in pyodide, set the environment
variable `PYODIDE_PACKAGES` to a comma separated list of packages. For
instance,
> 
> ```
> PYODIDE_PACKAGES="toolz,attrs" make
> ```
> Note that this environment variable must contain both the packages and their
> dependencies. The package names must match the folder names in `packages/`
> exactly; in particular they are case sensitive.
> 
> To build a minimal version of pyodide, set `PYODIDE_PACKAGES="micropip"`. The
micropip package is generally always included for any non empty value of
`PYODIDE_PACKAGES`.

Of course one can always build individual packages with `pyodide_build/buildpkg.py`, but the goal of this is to get a consistent but smaller pyodide distribution with a valid `packages.json`.

This also adds a minimal build with only `micropip` included. It should be fairly fast, and not make CI slower too much, since it would run in parallel with the main build job. I find it helpful to add this extra CI job, to see if we can optimize the size of files in this minimal distribution further.

cc @phorward

### CODE REVIEW

**Comment by phorward** [comment]
Thanks, that's cool!

Quite now, I'm using a [service script](https://github.com/mausbrand/pyodide-app-stub/blob/master/get-pyodide.py) which downloads the release tarball and "crops" the Pyodide-installation to a micropip-only version. The packages.json is built manually. This is usually not a good way, but a working one with minimal effort to get a Pyodide for our web-app development quickly.

**Comment by rth** [comment]
Good to know thanks @phorward ! By having a way to optional to build a minimal pyodide distribution, there is quite a few things we could optimize for the size (and load time) to the resulting pyodide package see https://github.com/iodide-project/pyodide/issues/646

**Comment by rth** [comment]
So with this option, the minimal build setup in CI takes 13min as opposed to 35min for the full pyodide build (assuming emsdk is already built).

File: pyodide_build/common.py [hunk 1]
```diff
@@ -24,3 +25,19 @@ def parse_package(package):
...
+
+    Returns:
+      a list of package names to build or None. The list may contain duplicate
+      values.
+    """
+    if query is None:
+        return None
+    packages = query.split(',')
+    packages = [el.strip() for el in packages]
+    return ['micropip', 'distlib'] + packages
```
[the remark this answers is not in the document]
Comment (rth) [reply pyodide_build/common.py]: Good point, the order doesn't indeed matter. Changed to a set.

**Comment by rth** [comment]
Thanks for the review @phorward ! I have addressed you comments.

(The chrome CI failure in scipy and mne can be ignored.)

**Comment by rth** [comment]
One thing I forgot to mention @phorward is that we should click on "squash and merge" button when merging, to get a single commit on master.

**Comment by phorward** [comment]
@rth oops okay I'm sorry ;-) next time I will do so! In the repository settings, direct merge can also be prohibited... 
image

**Comment by rth** [comment]
No worries. Once you do it that way once, I think it should automatically switch to it by default in your Github UI (or we can indeed disable it in setting).

**Comment by wlach** [comment]
> No worries. Once you do it that way, I think it should automatically switch to it by default in your Github UI (or we can indeed disable it in setting).

👍 on disabling merge commits, I'm honestly not sure why that option is still there...

**Pull request merged**